src/PrettyPrinting.jl

```diff
@@ -1546,6 +1546,8 @@ write_indent(io::IO) = write(_unwrap(io).io, io.indent_str^io.indent_level)
 
 write(io::IOCustom, chr::Char) = write(io, string(chr)) # Need to catch writing a '\n'
 
+write(io::IOCustom, chr::Symbol) = write(io, string(chr)) # Need to catch writing a '\n'
```

Comment:
This adds one invalidation. We can live with that for now. But I wonder if we shouldn't instead replace this and the one for char, and also `write(io::IOCustom, str::String)` by a single method for `unsafe_write(io::IOCustom, ref, nbytes::UInt)` which they all end up calling?

Reply:
Note that even if the data then is a byte stream, we don't have to worry about things like decoding UTF-8 or so -- the byte for `\n` is never part of a multibyte UTF-8 character.

Reply:
It would also allow removing
```
# Base.write on an IOContext does not call Base.write on the unwrapped context ...
Base.write(io::IOContext{<: IOCustom}, s::Union{SubString{String}, String}) = Base.write(Base.unwrapcontext(io)[1], s)
```
and maybe more, like 
```
Base.write(io::IOContext{<: IOCustom}, s::Char) = Base.write(Base.unwrapcontext(io)[1], s)
```

Conversely, if we don't provide `unsafe_write`, then perhaps this PR also needs `Base.write(io::IOContext{<: IOCustom}, s::Symbol) = Base.write(Base.unwrapcontext(io)[1], s)` ????


Anyway, now that I look at the code closer, I guess in order to implement `_write_line` right we need at least some UTF-8 handling after all... but I guess in the worste case, we could just convert the `Ptr{UInt8}` back into a `String` ;-)